include/fixed_containers/reflection.hpp

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#include "fixed_containers/concepts.hpp"
+#include "fixed_containers/fixed_string.hpp"
+#include "fixed_containers/fixed_vector.hpp"
+
+#include <array>
+#include <cassert>
+#include <string_view>
+#include <tuple>
+
+// https://clang.llvm.org/docs/LanguageExtensions.html#builtin-dump-struct
+static_assert(__has_builtin(__builtin_dump_struct),
+              "Static reflection requires __builtin_dump_struct() to be available");
+
+namespace fixed_containers::reflection_detail
```

Comment:
Is there going to be a wrapper around these functions in the `fixed_containers` namespace as well?

Reply:
Yes, in the `fixed_containers::reflection` namespace once it gets more stable.